misk/src/main/kotlin/misk/web/actions/ReadinessCheckAction.kt

```diff
@@ -37,20 +62,57 @@ class ReadinessCheckAction @Inject internal constructor(
       // Don't do healthchecks if services haven't all started. The app isn't in a good state yet,
       // and a health check could end up triggering random errors that we don't want to flood the
       // logs with.
-      return Response("", statusCode = 503)
+      return@runBlocking Response("", statusCode = 503)
+    }
+
+    // Only null on first run
+    // This will block since the refresh is being done on this thread
+    val (lastUpdate, statuses) = cachedStatus
+      ?: refreshStatuses()
+
+    // This needs to be independent of the max age check so that services can recover
+    // If refreshStatuses() is never run then a failed/stale check can never recover
+    if (
+      clock.instant().isAfter(lastUpdate.plusMillis(config.readiness_refresh_interval_ms.toLong()))
+      // don't launch concurrent refreshes
+      && refreshExecutor.activeCount == 0
+    ) {
```

Comment:
It might be easier to reason about the refresh logic if we packaged up the check status refresh into a recurring task and let a `RepeatedTaskQueue` manage the threading, and expose the most recent check result and timestamp on some singleton